app/src/test/java/net/squanchy/notification/UpcomingEventsServiceTest.kt

```diff
@@ -0,0 +1,135 @@
+
+    @Before
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        upcomingEventsService = UpcomingEventsService(
+            service,
+            currentTime,
+            INTERVAL
+        )
+    }
+
+    @Test
+    fun upcomingEventsEmitsOnlyEventsStartingAfterNowInsideInterval() {
+        givenServiceWillReturn(
+            BEFORE_NOW,
+            BEFORE_NOW_OTHER_TIMEZONE,
+            AFTER_NOW_INSIDE_INTERVAL,
+            AFTER_NOW_INSIDE_INTERVAL_OTHER_TIMEZONE,
+            AFTER_NOW_OUTSIDE_INTERVAL,
+            AFTER_NOW_OUTSIDE_INTERVAL_OTHER_TIMEZONE
+        )
+
+        upcomingEventsService.upcomingEvents().test()
+            .assertValue(listOf(
+                AFTER_NOW_INSIDE_INTERVAL,
+                AFTER_NOW_INSIDE_INTERVAL_OTHER_TIMEZONE
+            ))
+    }
+
+    @Test
+    fun nextEventsEmitsOnlyEventsStartingAfterNowOutsideInterval() {
+        givenServiceWillReturn(
+            BEFORE_NOW,
+            BEFORE_NOW_OTHER_TIMEZONE,
+            AFTER_NOW_INSIDE_INTERVAL,
+            AFTER_NOW_INSIDE_INTERVAL_OTHER_TIMEZONE,
+            AFTER_NOW_OUTSIDE_INTERVAL,
+            AFTER_NOW_OUTSIDE_INTERVAL_OTHER_TIMEZONE
+        )
+
+        upcomingEventsService.nextEvents().test()
+            .assertValue(listOf(
+                AFTER_NOW_OUTSIDE_INTERVAL,
+                AFTER_NOW_OUTSIDE_INTERVAL_OTHER_TIMEZONE
+            ))
+    }
+
+    private fun givenServiceWillReturn(vararg events: Event) {
+        Mockito.`when`(service.sortedFavourites())
+            .thenReturn(Single.just(events.asList()))
+    }
+
+    companion object {
+        private val USER_ZONE_ID = ZoneId.ofOffset("UTC", ZoneOffset.ofHours(0))
+        private val ONE_HOUR_EARLIER_ZONE_ID = ZoneId.ofOffset("UTC", ZoneOffset.ofHours(1))
+
+        private val NOW = ZonedDateTime.of(
+            2018,
+            12,
+            20,
+            11,
+            25,
+            0,
+            0,
+            USER_ZONE_ID
+        )
+
+        val INTERVAL: Duration = Duration.ofMinutes(30)
+
+        val BEFORE_NOW = anEvent(
+            title = "Before now",
+            startTime = NOW.minusHours(1).toLocalDateTime(),
+            endTime = NOW.plusHours(4).toLocalDateTime(),
+            timeZone = USER_ZONE_ID
+        )
+
+        val BEFORE_NOW_OTHER_TIMEZONE = anEvent(
+            title = "Before now other timezone",
+            startTime = NOW.toLocalDateTime().plusMinutes(30),
+            endTime = NOW.plusHours(4).toLocalDateTime(),
+            timeZone = ONE_HOUR_EARLIER_ZONE_ID
+        )
+
+        val AFTER_NOW_INSIDE_INTERVAL = anEvent(
+            title = "After now inside interval",
+            startTime = NOW.plus(INTERVAL.dividedBy(2)).toLocalDateTime(),
+            endTime = NOW.plusHours(4).toLocalDateTime(),
+            timeZone = USER_ZONE_ID
+        )
+        val AFTER_NOW_INSIDE_INTERVAL_OTHER_TIMEZONE = anEvent(
```

Comment:
Nit: whiteline before this

Reply:
f49a024127f726d1d60826afccab675646644080